### PR TITLE
RDKB build support

### PR DESCRIPTION
**Pull request by vitalybu**
Add support for out-of-tree RDKB build

### CODE REVIEW

File: tools/cmake/toolchain/rdkb.cmake [hunk 1]
```diff
@@ -0,0 +1,80 @@
+## General configuration
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_PROCESSOR ARMV7 CACHE STRING "System Processor")
```
Comment (tomereli) [inline tools/cmake/toolchain/rdkb.cmake]: Do we need it to be ARMV7?

File: tools/cmake/toolchain/rdkb.cmake [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+set(PLATFORM_INCLUDE_DIR        ${PLATFORM_STAGING_DIR}/usr/include CACHE STRING "Platform Include Directory")
+set(ENV{PKG_CONFIG_PATH}        "${PLATFORM_STAGING_DIR}/usr/lib/pkgconfig")
+
+# Compiler Definitions
+add_definitions(
+    -DBEEROCKS_RDKB
+)
+
+## Library config
+set(BUILD_SHARED_LIBS ON CACHE BOOL "BUILD_SHARED_LIBS" FORCE)
```
Comment (tomereli) [inline tools/cmake/toolchain/rdkb.cmake]: I recall that @arnout said these are not needed.

File: cmake/multiap-helpers.cmake [hunk 3]
```diff
@@ -18,6 +18,17 @@ if(NOT WIN32)
   set(BoldWhite   "${Esc}[1;37m")
 endif()
 
+# Force the compiler to produce ANSI-colored output when using ninja build
```
Comment (tomereli) [inline cmake/multiap-helpers.cmake:21]: Nice!

**Review by tomereli — APPROVED** [review]

**Review by morantr — APPROVED** [review]

File: tools/cmake/toolchain/rdkb.cmake (hunk 1 above)
Comment (vitalybu) [reply tools/cmake/toolchain/rdkb.cmake]: Nope, I'll remove it

File: tools/cmake/toolchain/rdkb.cmake (hunk 2 above)
Comment (vitalybu) [reply tools/cmake/toolchain/rdkb.cmake]: Will remove that as well

File: tools/commands/build.py [hunk 4]
```diff
@@ -196,9 +196,12 @@ def __init__(self, args):
         modules_dir = os.path.realpath(args.map_path)
 
         map_cmake_flags = args.cmake_flags
-        if not args.native:
-            cmake_toolchain_file = os.environ.get('CMAKE_TOOLCHAIN_FILE', 'external_toolchain.cmake')
-            map_cmake_flags = ["CMAKE_TOOLCHAIN_FILE=" + cmake_toolchain_file] + map_cmake_flags
+        if not args.native: 
+            if "CMAKE_TOOLCHAIN_FILE" in os.environ:
+                map_cmake_flags += ["CMAKE_TOOLCHAIN_FILE=" + os.path.abspath(os.environ.get('CMAKE_TOOLCHAIN_FILE'))]
```
Comment (arnout) [inline tools/commands/build.py:201]: Weird, I was convinced I had merged this exact change some time ago...

**Review by arnout — CHANGES_REQUESTED** [review]
The Dockerfile is not the approach we want to take, because it still requires you to have the whole RDKB build on the host, so there is not much of an advantage of building it in Docker. Worse, if your host is not Ubuntu 18.04, it may fail in weird ways. See #545 for the proper approach. Note that #545 is not finished yet. So I'm OK with merging this RDKB Dockerfile as is for the moment; just be warned that I'm planning to kill it :-)

Do not make merge commits; rebase on master instead of merging it.

Otherwise LGTM. Still changes requested because of the merge commit.

**Comment by vitalybu** [comment]
> The Dockerfile is not the approach we want to take, because it still requires you to have the whole RDKB build on the host, so there is not much of an advantage of building it in Docker. Worse, if your host is not Ubuntu 18.04, it may fail in weird ways. See #545 for the proper approach. Note that #545 is not finished yet. So I'm OK with merging this RDKB Dockerfile as is for the moment; just be warned that I'm planning to kill it :-)
> 
> Do not make merge commits; rebase on master instead of merging it.
> 
> Otherwise LGTM. Still changes requested because of the merge commit.

I agree that having the entire toolchain and staging directory inside a docker is a much better approach.
The purpose of this dockerfile is to remove the dependency on the build host (for example, one can run an Ubuntu 18.04 docker on Ubuntu 14.04 without issues).

As for the merge commit, it was automatically applied by GitHub once it detected that the branch is out of date (with a press of a button), but I'll remove it

File: tools/commands/build.py (hunk 4 above)
Comment (vitalybu) [reply tools/commands/build.py:201]: ¯\\_(ツ)_/¯

**Review by arnout — APPROVED** [review]

**Pull request merged**